google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/CloudApiMavenService.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.apis;
+
+import com.google.common.collect.ImmutableList;
+import com.intellij.jarRepository.JarRepositoryManager;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.diagnostic.Logger;
+import java.util.Comparator;
+import java.util.List;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.version.Version;
+import org.jetbrains.idea.maven.aether.ArtifactKind;
+import org.jetbrains.idea.maven.aether.ArtifactRepositoryManager;
+import org.jetbrains.idea.maven.aether.ProgressConsumer;
+
+/** An application service providing Cloud API Maven functionality. */
+public class CloudApiMavenService {
+  private static final Logger logger = Logger.getInstance(CloudApiManager.class);
+
+  private static final String GOOGLE_CLOUD_JAVA_BOM_GROUP_NAME = "com.google.cloud";
+  private static final String GOOGLE_CLOUD_JAVA_BOM_ARTIFACT_NAME = "google-cloud-bom";
+  private static final String GOOGLE_CLOUD_JAVA_BOM_VERSION_CONSTRAINT = "[0,)";
```

Comment:
Interesting math notation, is this what they used in Aether? Is this required or we can fetch all versions regardless of constraint?

Reply:
yes this is Aether notation.

The service call requires a version to be passed in. So we must pass in some pattern to indicate that we want them all. I toyed with the idea of using this notation itself to limit the results instead of limiting them later client-side, but we would have to pre-suppose a version syntax (which is a bad idea I think).